pkg/resources/virtual/types/types.go

```diff
@@ -0,0 +1,17 @@
+package types
+
+// VirtualTransformFunc is a cache.TransformFunc. Re-implemented here so that we can implement a chain method for UX.
+// When implementing, note that you need to explicitly handle signal types of objects like cache.DeletedFinalStateUnknown
+type VirtualTransformFunc func(any) (any, error)
+
+// Chain combines this VirtualTransformFunction with the provided VirtualTransformFunc. The returned function will call
+// v then other. If v returns an error, other will not be called.
+func (v VirtualTransformFunc) Chain(other VirtualTransformFunc) VirtualTransformFunc {
+	return func(a any) (any, error) {
+		newObj, err := v(a)
+		if err != nil {
+			return nil, err
+		}
+		return other(newObj)
+	}
+}
```

Comment:
Technically we don't need this code. My thinking was that this was useful if we have a type which has virtual fields specific to it. I think there's already a UI request for such fields: rancher/rancher#40771, so it's not thinking too far ahead. 

However, I think that you are strictly correct that we don't need this right now. I'll defer to @ericpromislow on this one - if the team prefers to remove it I can do that.

Reply:
I go with removing this code. From what I can tell it's only used in `TestChain`, so I assume there's no need for that test either. All we then have is a definition of `VirtualTransformFunc`, but this looks like the right place for it.

Reply:
Since @ericpromislow prefers removing it, I can . @ericpromislow keep in mind that I will also remove the VirtualTransformFunc, since the point of that type was to allow implementing this method.